test/zzzz-duplicate-entries.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# This is a test suite for testing basic log processing
+# functionality and existance of duplicate records when journal rolls over
+
+source "$(dirname "${BASH_SOURCE[0]}" )/../hack/lib/init.sh"
+source "${OS_O_A_L_DIR}/hack/testing/util.sh"
+os::util::environment::use_sudo
+
+os::test::junit::declare_suite_start "test/zzzz-duplicate-entries"
+
+es_pod=$( get_es_pod es )
+es_ops_pod=$( get_es_pod es-ops )
+es_ops_pod=${es_ops_pod:-$es_pod}
+
+cleanup() {
+    local return_code="$?"
+    set +e
+
+    if [ -f /etc/systemd/journald.conf.bak ]; then
+      sudo mv /etc/systemd/journald.conf.bak /etc/systemd/journald.conf
+      sudo systemctl restart systemd-journal
+    fi
+
+    # cleanup fluentd pos file and restart
+    flush_fluentd_pos_files
+    oc label node --all logging-infra-fluentd=true 2>&1 | artifact_out
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+
+    # this will call declare_test_end, suite_end, etc.
+    os::test::junit::reconcile_output
+    exit $return_code
+}
+trap "cleanup" EXIT
+
+# turn off fluentd
+oc label node --all logging-infra-fluentd- 2>&1 | artifact_out || :
+os::cmd::try_until_text "oc get daemonset logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $((second * 120))
+
+sudo sed -i.bak "s/^.MaxRetentionSec=.*$/ MaxRetentionSec=30/" /etc/systemd/journald.conf
+sudo systemcl restart systemd-journal
+
+get_logmessage() {
+    logmessage="$1"
+}
+get_logmessage2() {
+    logmessage2="$1"
+}
+sleep 10
+# we're actually waiting for rsyslog instead . . .
+wait_for_fluentd_to_catch_up get_logmessage get_logmessage2
```

Comment:
You don't need to use the `get_logmessage get_logmessage2` parameters and functions if you don't need the messages created by `wait_for_fluentd_to_catch_up`

Reply:
actually, this will always error out - because fluentd is shutdown - I think you need to restart fluentd after restarting the journal, then call `wait_for_fluentd_to_catch_up`